core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/EventConsumer.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.openwhisk.core.monitoring.metrics
+
+import java.lang.management.ManagementFactory
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.kafka.ConsumerMessage.CommittableOffsetBatch
+import akka.kafka.scaladsl.Consumer
+import akka.kafka.scaladsl.Consumer.DrainingControl
+import akka.kafka.{ConsumerSettings, Subscriptions}
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Keep, Sink}
+import javax.management.ObjectName
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import kamon.Kamon
+import kamon.metric.MeasurementUnit
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+import org.apache.openwhisk.core.connector.{Activation, EventMessage, Metric}
+
+trait MetricRecorder {
+  def processActivation(activation: Activation, initiatorNamespace: String): Unit
+  def processMetric(metric: Metric, initiatorNamespace: String): Unit
+}
+
+case class EventConsumer(settings: ConsumerSettings[String, String], recorders: Seq[MetricRecorder])(
+  implicit system: ActorSystem,
+  materializer: ActorMaterializer) {
+  import EventConsumer._
+
+  private implicit val ec: ExecutionContext = system.dispatcher
+
+  //Record the rate of events received
+  private val activationCounter = Kamon.counter("openwhisk.userevents.global.activations")
```

Comment:
Should we use the same metrics naming convention used in invoker/controller?
e.g.
```
#kamon related configuration
kamon {
    environment {
      # Identifier for this service. For keeping it backward compatible setting to natch previous
      # statsd name
      service = "openwhisk-statsd"
    }
```
then metric names like "global.activations"